packages/@glimmer/syntax/lib/generation/print.ts

```diff
@@ -57,7 +57,7 @@ export default function build(ast) {
     }
     break;
     case 'MustacheCommentStatement': {
-      output.push(compactJoin(['{{!', ast.value, '}}']));
+      output.push(compactJoin(['{{!--', ast.value, '--}}']));
```

Comment:
Agree that the type of comment should be added to the AST. However, this is absolutely a bug, because if you do have curlies in a comment, which is totally feasible and super normal use case, it results in a parse error. My fix is a valid fix for the bug. We can change it to be more "accurate" with the comments once the AST has more information.